pkg/distribution/client.go

```diff
@@ -172,7 +172,10 @@ func (c *Client) PullModel(ctx context.Context, reference string, progressWriter
 		// Report progress for local model
 		if progressWriter != nil {
 			size := fileInfo.Size()
-			fmt.Fprintf(progressWriter, "Using cached model: %.2f MB\n", float64(size)/1024/1024)
+			err := writeSuccess(progressWriter, fmt.Sprintf("Using cached model: %.2f MB", float64(size)/1024/1024))
+			if err != nil {
+				c.log.Warnf("Writing progress: %v", err)
```

Comment:
Should we set `progressWriter = nil` in this case (and in similar cases below)?  I think the JSON stream might be corrupted at that point anyway if the write fails.